SW-2300 detect air filter iii in mr beam plugin

JFSolution opened the pull request:
- detect air filter in Mr Beam Plugin
- switch profile of the air filter
- use profiles for maintenance parameters
- use profiles for maintenance page
- fix shopify links

Commit:
-add shopify link to profiles and fix profile id 1

Commit:
add profile handling to airfilter class

Commit:
- get lifespan and shopify links for all filterstages of the given filter
- get lifespan and shopify links in frontend
- added more testing
- fix broken shopify links

Commit:
fix codesmell

irlaec commented on octoprint_mrbeam/profiles/airfilter_system/airfilter_system_id_8.yaml:
```diff
@@ -1,6 +1,11 @@
-  - lifespan: 40 #TODO
-carbon_filter:
-  - lifespan: 280 #TODO
+  - lifespan: 80 #TODO
```
Could you please add a explanation to these TODO and maybe a ticket number?

JFSolution replied:
sure

irlaec commented on octoprint_mrbeam/static/js/app/view-models/settings/maintenance.js:
```diff
@@ -356,6 +353,34 @@ $(function () {
+            } else {
+                link = null;
+            }
+            if (link != null) {
```
Wouldn't be enough by just returning link? We already assign it a null value if it doesn't have one.
I think we could just remove this second if-else

JFSolution replied:
You are right

irlaec commented on tests/iobeam/test_aifilter.py:
```diff
@@ -1,11 +1,28 @@
-from octoprint_mrbeam.iobeam.airfilter import AirFilter
+from octoprint_mrbeam.iobeam.airfilter import AirFilter, airfilter
+
+pluginmock = MagicMock(
```
We have a fixture for the MrBeamPlugin (in conftest), I don't know if it would make sense to use that instead

JFSolution replied:
I changed it to use this instead, thanks for the tip

Commit:
adding PR feedback:
- adding ticket for todo
- remove not needed if condition
- use fixture to mock the plugin

Commit:
fix conftest

irlaec reviewed (approved):
I just noticed the name of the test file has a typo --> test_aiRfilter.py

Commit:
fix typo in filename
